src/key_value_store/core/KeyValueStore.cpp

```diff
@@ -39,4 +39,24 @@ bool KeyValueStore::Remove(const std::string &key) {
   const std::lock_guard<std::mutex> lock(lock_);
   result = kv_store_.erase(key);
   return result;
+}
+
+std::vector<std::pair<std::string, std::vector<std::string>>>
+KeyValueStore::GetAllEntries() {
+  std::vector<std::pair<std::string, std::vector<std::string>>> toReturn;
+  const std::lock_guard<std::mutex> lock(lock_);
+  for (const std::pair<std::string, std::vector<std::string>> &pair :
```

Comment:
Hahaha I personally don't really like using auto because I feel like knowing exactly what the iterable is helps in understanding the code. If you think its a big issue let me know and I'll be happy to change it!

Reply:
definitely not a big deal, I think it is perfectly fine without auto